lib/index.js

```diff
@@ -86,14 +86,19 @@ Optimizely.prototype.initialize = function() {
  */
 
 Optimizely.prototype.track = function(track) {
+  var opts = this.options;
   // Use the new-style API (which is compatible with Classic and X)
   var payload = {
     type: 'event',
     eventName: track.event(),
     tags: track.properties()
   };
 
-  if (payload.tags.revenue) {
+  if (payload.tags.revenue && opts.sendRevenueOnlyForOrderCompleted) {
```

Comment:
There is still a bug with revenue sent to the SDK later not getting the same logic applied. Can this be moved before the payload so that both services are formatted the same?